File: ProjectEulerPython/problems/problem_001.py
from problems.problem import Problem

class Problem001(Problem):

  def calculate_answer(self) -> int:
    answer = 0
    n = 1000
    self.print_detail("Multiples:")
    for i in range(1, n):
      if i % 3 == 0 or i % 5 == 0:
        self.print_detail(i)
        answer += i
    return answer

File: ProjectEuler/Problems/Problem001.cs
using Microsoft.Extensions.Logging;

namespace ProjectEuler.Problems
{
    public class Problem001 : Problem<int>
    {
        public Problem001(ILogger<Problem001> logger) : base(logger) { }

        protected override int CalculateAnswer()
        {
            Logger.LogDebug("Multiples:");
            var answer = 0;
            for (var i = 1; i < 1000; ++i)
            {
                if (i % 3 == 0 || i % 5 == 0)
                {
                    Logger.LogDebug("{multiple}", i);
                    answer += i;
                }
            }
            return answer;
        }
    }
}
